src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,74 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { useDispatch } from 'react-redux';
+import { currentTodoReducer } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { useAppSelector } from '../../app/hooks';
 
 export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+  const dispatch = useDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const handleClose = () => {
+    dispatch(currentTodoReducer(null));
+  };
 
-      <Loader />
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  useEffect(() => {
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .catch(() => {
+          throw new Error("Can't get User information");
+        })
+        .finally(() => setIsLoading(false));
+    }
+  }, [currentTodo]);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+  return (
+    !!currentTodo && (
```

Comment:
better to add the condition before the jsx
```suggestion
 if(!currentTodo) {
   return;
 }

  return (
```